src/v0/destinations/klaviyo/transform.js

```diff
@@ -275,9 +301,101 @@ const process = async (event) => {
   return result;
 };
 
+const batchEvents = (successRespList) => {
+  const batchedResponseList = [];
+  const identifyResponseList = [];
+  successRespList.forEach((event) => {
+    const processedEvent = event;
+    if (processedEvent.message.length === 2) {
+      identifyResponseList.push(event.message[0]);
+      [processedEvent.message] = event.message.slice(1);
+    } else {
+      [processedEvent.message] = processedEvent.message;
+    }
+  });
+  const subscribeEventGroups = _.groupBy(
+    successRespList,
+    (event) => event.message.body.JSON.data.attributes.list_id,
+  );
+  Object.keys(subscribeEventGroups).forEach((listId) => {
+    // eventChunks = [[e1,e2,e3,..batchSize],[e1,e2,e3,..batchSize]..]
+    const eventChunks = _.chunk(subscribeEventGroups[listId], MAX_BATCH_SIZE);
+    eventChunks.forEach((chunk) => {
+      const batchEventResponse = generateBatchedPaylaodForArray(chunk);
+      batchedResponseList.push(
+        getSuccessRespEvents(
+          batchEventResponse.batchedRequest,
+          batchEventResponse.metadata,
+          batchEventResponse.destination,
+          true,
+        ),
+      );
+    });
+  });
+  identifyResponseList.forEach((response) => {
+    batchedResponseList[0].batchedRequest.push(response);
+  });
+  return batchedResponseList;
+};
+
+// This function separates subscribe response and other responses in chunks
+const getEventChunks = (event, subscribeRespList, otherRespList) => {
+  if (Array.isArray(event.message)) {
+    subscribeRespList.push(event);
```

Comment:
Is there any other safe way rather than checking of its Array only for Subscribed Resp List?